lib/compress/zstd_lazy.c

```diff
@@ -261,6 +339,10 @@ static size_t ZSTD_DUBT_findBestMatch (
 
         *smallerPtr = *largerPtr = 0;
 
+        if (dictMode == ZSTD_dictMatchState && nbCompares) {
```

Comment:
Should we give the dictionary a fixed lower limit on the number of compares, like at least 1 compare, or at least 8 compares? Or have benchmarks to show that it doesn't matter?

Reply:
The short answer is, I haven't experimented with those kinds of changes. The scope of these changesets has been to produce identical behavior faster.

So yeah, there are probably opportunities to experiment with different strategies given the additional information available. I'd love to chat about how to go about doing that.

Reply:
I agree this seems like an additional topic.

This new strategy is likely to alter the speed/compression trade-off, spending more cpu for more compression opportunity, in a way which is probably content specific.
It would have to be tested against a corpus to observe the impact.

The nice thing though is that it seems relatively easy to test.
Just add a `nbCompares = MAX(nbCompares, CONSTANT);` before the dictionary section, and that's it from a code perspective. A more important question is the selection of a benchmark corpus, though Managed Compression might help on this front.